src/components/DateSelect/DateSelect.tsx

```diff
@@ -0,0 +1,234 @@
+  to: string
+  from: string
+}
+
+const DateSelectOption = (props: SelectOptionProps<DateValue>) => {
+  return <Select.Option {...props} />
+}
+
+export interface DateSelectProps {
+  /** The id of the DateSelectProps. Use this prop to make label and `helperText` accessible for screen readers */
+  id?: SelectProps["id"]
+  /** Label text to be placed before the element */
+  label?: SelectProps["label"]
+  /** The placeholder text, shown when there is no selected value */
+  placeholder?: SelectProps["placeholder"]
+  /** The value of the `select` element, required for a controlled component */
+  value?: DateValue
+  /** Default value which will not trigger the `filled` select state */
+  defaultValue?: DateValue
+  /** Whether the component is disabled or not */
+  disabled?: SelectProps["disabled"]
+  /** Callback fired when the value is changed */
+  onChange?: (changes: DateValue) => void
+  /** Highlight the select when it's in a `filled` state */
+  highlightFilled?: SelectProps["highlightFilled"]
+  /** Select grows to fill the width of the parent */
+  fullWidth?: SelectProps["fullWidth"]
+  children: SelectProps["children"]
+  /** Format in which the date should be in, defaults to 'MM/dd/yyyy' */
+  dateFormat?: string
+}
+
+const DateSelect = ({
+  id,
+  label,
+  placeholder,
+  value,
+  defaultValue,
+  disabled = false,
+  onChange,
+  highlightFilled = true,
+  fullWidth = false,
+  children,
+  dateFormat = "MM/dd/yyyy",
+}: DateSelectProps) => {
+  const [fromValue, setFromValue] = useState("")
+  const [toValue, setToValue] = useState("")
+
+  const isDateFromValid = useMemo(() => {
+    if (!fromValue) return true
+    const dateFrom = parse(fromValue, dateFormat, new Date())
+    return isValid(dateFrom)
+  }, [dateFormat, fromValue])
+
+  const isDateToValid = useMemo(() => {
+    if (!toValue) return true
+    const dateTo = parse(toValue, dateFormat, new Date())
+    return isValid(dateTo)
+  }, [dateFormat, toValue])
+
+  const isFormValid = useMemo(() => {
+    const dateFrom = parse(fromValue, dateFormat, new Date())
+    const dateTo = parse(toValue, dateFormat, new Date())
+    if (!isValid(dateFrom) || !isValid(dateTo) || dateFrom > dateTo)
+      return false
+
+    return true
+  }, [dateFormat, fromValue, toValue])
+
+  const [customValue, setCustomValue] = useState({ to: "", from: "" })
+
+  const resetValue = useCallback(() => {
+    setFromValue(customValue.from)
+    setToValue(customValue.to)
+  }, [customValue.from, customValue.to])
+
+  return (
+    <Select
+      id={id}
+      label={label}
+      placeholder={placeholder}
+      value={value as SelectProps["value"]}
+      defaultValue={defaultValue as SelectProps["defaultValue"]}
+      disabled={disabled}
+      onChange={(value) => {
+        if (value) onChange?.((value as unknown) as DateValue)
+      }}
+      highlightFilled={highlightFilled}
+      fullWidth={fullWidth}
+    >
+      {children}
+      <SelectOptionCustom
```

Comment:
This custom option can be optional, default rendered but removed with some flag.

Reply:
Added option in cd14c96